src/mixins/dropdown.js

```diff
@@ -346,17 +321,31 @@ export default {
       // Tab, if in a text-like input, we should just focus next item in the dropdown
       // Note: Inputs are in a special .dropdown-form container
     },
-    onFocusOut (evt) {
-      if (this.$el.contains(evt.relatedTarget)) {
-        return
-      }
-      setTimeout(() => {
-        this.visible = false
-      })
-    },
     onMouseOver (evt) /* istanbul ignore next: not easy to test */ {
       // Removed mouseover focus handler
     },
+    isClickOut () {
+      return true
```

Comment:
I think this is why clicking on non-interactive elements in a menu closes the menu.

Shouldn't this be a check to see if the event.target is not contained in menu container (and return true if it is, otherwise return false)?  As the purpose of clickout is to determine if something _outside_ of the menu was clicked (and close it)

Reply:
Although, technically, clicking outside of the menu _should_ move focus from the menu item to the clicked element (or the documentElement if not clicking on an interactive element).  So the focus handler _should_ (I would assume) close the menu (unless some browsers don't move focus to the document, possibly Mobile).

Reply:
What I did here was to use the "click out" listener to replace the "close on dropdown item click" logic. But this caused some issues you have correctly pointed out.

> Although, technically, clicking outside of the menu should move focus from the menu item to the clicked element (or the documentElement if not clicking on an interactive element). So the focus handler should (I would assume) close the menu (unless some browsers don't move focus to the document, possibly Mobile).
Just clicking out of the menu doesn't fire the `focusin` handler when the next element isn't interactive. This is why I added back the "click out" listener.

With the combination of the current listener setup it seems to work very well.

Reply:
That must be a safari issue with respect to not firing a focusout/in on clicking the document (I kinda recall reading about that issue somewhere on the web... maybe quirksmode, but I thought it might be safari mobile only)

Reply:
For me it was the same on Chrome and Firefox.